### PR TITLE
loadConfigFromFile no longer logs an error if file doesn't exist

### PR DESCRIPTION
**Pull request by wmathurin**
It returns the error to the caller - caller that cares should log an error

### CODE REVIEW

File: libs/SmartStore/SmartStore/Classes/SFSDKStoreConfig.m [hunk 1]
```diff
@@ -44,7 +44,7 @@ @implementation SFSDKStoreConfig
 - (nullable id)initWithResourceAtPath:(NSString *)path {
     self = [super init];
     if (self) {
-        NSDictionary *config = [SFSDKResourceUtils loadConfigFromFile:path];
+        NSDictionary *config = [SFSDKResourceUtils loadConfigFromFile:path error:nil];
         self.soupsConfig = config == nil ? nil : config[kStoreConfigSoups];
```
Comment (wmathurin) [inline libs/SmartStore/SmartStore/Classes/SFSDKStoreConfig.m]: Code does not always expect a config - so if no config found, we don't need to log any error.

**Comment by wmathurin** [comment]
@bhariharan 
> Are we okay on semver? ;-)

=> Adressed with https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2858/commits/6208b3f9a6bcdcb5d57b3b86379a06bcd4d03ce2

**Comment by bhariharan** [comment]
Thanks, @wmathurin. Maybe we should mark the old one deprecated and remove it in `8.0`?

**Comment by wmathurin** [comment]
@bhariharan Doing it here: https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2860

**Pull request merged**